data/users/externalldap.py

```diff
@@ -53,26 +53,19 @@ def __init__(
         allow_tls_fallback=False,
         timeout=None,
         network_timeout=None,
-        bytes_mode=None,
     ):
         self._ldap_uri = ldap_uri
         self._user_dn = user_dn
         self._user_pw = user_pw
         self._allow_tls_fallback = allow_tls_fallback
         self._timeout = timeout
         self._network_timeout = network_timeout
-        # Currently python-ldap defaults this option to True for backward compatibility,
-        # which _will_ be deprecated once Python 2 reaches EOL.
-        # https://www.python-ldap.org/en/python-ldap-3.3.0/bytes_mode.html#bytes-mode
-        self._bytes_mode = bytes_mode or False
         self._conn = None
 
     def __enter__(self):
         trace_level = 2 if os.environ.get("USERS_DEBUG") == "1" else 0
 
-        self._conn = ldap.initialize(
-            self._ldap_uri, trace_level=trace_level, bytes_mode=self._bytes_mode
-        )
+        self._conn = ldap.initialize(self._ldap_uri, trace_level=trace_level)
```

Comment:
Does `bytes_mode` affect the cert issue? If not, it might be better to leave that option there, given that it will default to True otherwise. 
> If not given explicitly, python-ldap will default to bytes_mode=True, but if a unicode value is supplied to it, it will warn and use that value.

Reply:
I guess it's not going to have an effect in Py3, given that all str are unicode.

Reply:
bytes_mode cannot be True in python3. Just legacy cruft.